internal/rbd/rbd_healer.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ceph/ceph-csi/internal/util"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+// accessModeStrToInt convert access mode type string to int32.
+// Makesure to update this function as and when there are new modes introduced.
+func accessModeStrToInt(mode v1.PersistentVolumeAccessMode) csi.VolumeCapability_AccessMode_Mode {
+	switch mode {
+	case v1.ReadWriteOnce:
+		return csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER
+	case v1.ReadOnlyMany:
+		return csi.VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY
+	case v1.ReadWriteMany:
+		return csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER
+	}
+	return csi.VolumeCapability_AccessMode_UNKNOWN
+}
+
+func getSecret(c *k8s.Clientset, ns, name string) (map[string]string, error) {
+	deviceSecret := make(map[string]string)
+
+	secret, err := c.CoreV1().Secrets(ns).Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		util.ErrorLogMsg("get secret failed, err: %v", err)
+		return nil, err
+	}
+
+	for k, v := range secret.Data {
+		switch k {
+		case util.CredUserID, util.CredUserKey, util.EncryptionPassphraseKey:
+			deviceSecret[k] = string(v)
+		default:
```

Comment:
It is possible to store additional key/values in the secret, for example for the Vault connection (KMS-provider `vault`). Is there a good reason to filter keys? If filtering is needed, it should probably be done with a utility function so that it can be maintained elsewhere and additional keys added (and then you don't need to export these keys).

Reply:
Hmm, I don't recollect now. But mostly it was from my understanding.

Removing the filter, let's catch if anything is seen from CI results.